src/Microsoft.Azure.SignalR.Common/RestClients/RestClientBuilder.cs

```diff
@@ -12,22 +12,26 @@ internal class RestClientBuilder
     {
         private readonly Uri _baseUri;
         private readonly ServiceClientCredentials _credentials;
-        private DelegatingHandler[] _handlers;
+        private readonly DelegatingHandler[] _handlers;
         private HttpClientHandler _rootHandler;
 
-        public RestClientBuilder(ServiceEndpoint endpoint)
+        public RestClientBuilder(ServiceEndpoint endpoint, string userAgent) : this(userAgent)
         {
             _baseUri = new Uri(endpoint.Endpoint);
             _credentials = new JwtTokenCredentials(endpoint.AccessKey);
         }
 
-        public RestClientBuilder(string connectionString) : this(new ServiceEndpoint(connectionString)) { }
+        public RestClientBuilder(string connectionString, string userAgent) : this(new ServiceEndpoint(connectionString), userAgent) { }
 
-        public RestClientBuilder WithUserAgent(string usegAgent)
+        private RestClientBuilder(string userAgent)
         {
-            var azUserAgentHandler = new AsrsUserAgentHandler(usegAgent);
+            if (string.IsNullOrWhiteSpace(userAgent))
+            {
+                throw new ArgumentException($"'{nameof(userAgent)}' cannot be null or whitespace", nameof(userAgent));
+            }
+
+            var azUserAgentHandler = new AsrsUserAgentHandler(userAgent);
             _handlers = new DelegatingHandler[] { azUserAgentHandler };
-            return this;
         }
 
         internal RestClientBuilder WithRootHandler(HttpClientHandler rootHandler)
```

Comment:
public

Reply:
This method is supposed to be internal when the class is public in the future.